### PR TITLE
exculde hidden culomns

### PR DESCRIPTION
**Pull request by itzharDev**
https://github.com/parse-community/parse-dashboard/issues/1693

### CODE REVIEW

File: src/dashboard/Data/Browser/Browser.react.js [hunk 1]
```diff
@@ -355,6 +356,14 @@ class Browser extends DashboardView {
     }
 
     query.limit(MAX_ROWS_FETCHED);
+    let columns = localStorage.getItem(`ParseDashboard:${VERSION}:${this.props.params.appId}:${source}`)
```
Comment (davimacedo) [inline src/dashboard/Data/Browser/Browser.react.js:359]: I believe we should get it from `ColumnPreferences.getOrder` like in here: https://github.com/parse-community/parse-dashboard/blob/master/src/dashboard/Data/Browser/DataBrowser.react.js#L28

**Review by davimacedo — CHANGES_REQUESTED** [review]

File: src/dashboard/Data/Browser/Browser.react.js (hunk 1 above)
Comment (itzharDev) [reply src/dashboard/Data/Browser/Browser.react.js:359]: I already tried this.
there are missing params at the query point to use it

Comment (davimacedo) [reply src/dashboard/Data/Browser/Browser.react.js:359]: They should be all available at the DataBrowser component. Have you tried to send them up?

Comment (itzharDev) [reply src/dashboard/Data/Browser/Browser.react.js:359]: tried, yes. there is no data anywhere at query point

    console.log('this.refs.dataBrowser', this.refs.dataBrowser);
    console.log('this.state', this.state);
    console.log('this.props', this.props);


aaa

Comment (itzharDev) [reply src/dashboard/Data/Browser/Browser.react.js:359]: tried, yes. there is no data anywhere at query point

    console.log('this.refs.dataBrowser', this.refs.dataBrowser);
    console.log('this.state', this.state);
    console.log('this.props', this.props);

aaa

Comment (davimacedo) [reply src/dashboard/Data/Browser/Browser.react.js:359]: I beleive it should be added in here: https://github.com/parse-community/parse-dashboard/blob/master/src/dashboard/Data/Browser/Browser.react.js#L166

In the same way it is currently doing for the ordering state.

Comment (itzharDev) [reply src/dashboard/Data/Browser/Browser.react.js:359]: if you talk about `getColumnSort` it gives only a single column which class sort by

Comment (davimacedo) [reply src/dashboard/Data/Browser/Browser.react.js:359]: In a similar way you could use the `ColumnPreferences.getOrder` function in that point in order to get the visible columns and set it to the component state.

Comment (itzharDev) [reply src/dashboard/Data/Browser/Browser.react.js:359]: tried and not do the work

    ColumnPreferences.getOrder(
          {},
          this.props.params.appId,
          source
        ))

result `[]`

Comment (davimacedo) [reply src/dashboard/Data/Browser/Browser.react.js:359]: Try with:
```js
   const columnPreferences = context.currentApp.columnPreference || {}
    let order = ColumnPreferences.getOrder(
      {},
      context.currentApp.applicationId,
      props.className,
      columnPreferences[className]
    );
```

Comment (itzharDev) [reply src/dashboard/Data/Browser/Browser.react.js:359]: same

Comment (davimacedo) [reply src/dashboard/Data/Browser/Browser.react.js:359]: it is probably the first parameter. Instead of `{}`, try with `this.getColumns(props.className)`

Comment (itzharDev) [reply src/dashboard/Data/Browser/Browser.react.js:359]: 1. I don't find `getColumns` in all project
2. I've try also this:

Screen Shot 2021-05-04 at 8 43 41 PM


Screen Shot 2021-05-04 at 8 43 51 PM

Comment (davimacedo) [reply src/dashboard/Data/Browser/Browser.react.js:359]: Sorry. It is `getClassColumns`

Comment (itzharDev) [reply src/dashboard/Data/Browser/Browser.react.js:359]: its also miss some data:

Screen Shot 2021-05-04 at 8 57 26 PM

Screen Shot 2021-05-04 at 8 57 09 PM

Comment (davimacedo) [reply src/dashboard/Data/Browser/Browser.react.js:359]: It is in the right direction but you will have to figure it out. Probably, when the schema is not fetched yet, we should pass nothing in the first parameter or only consider `columnPreferences[props.className]`. Btw did you try to log the content of this?

Comment (itzharDev) [reply src/dashboard/Data/Browser/Browser.react.js:359]: I'm working on it locally but connected to my production MongoDB
probably, I build the project from my own branch and use it. if you take my solution and fix it - will be community profit

Comment (itzharDev) [reply src/dashboard/Data/Browser/Browser.react.js]: ok, I found the right way get columns
code updated

**Review by davimacedo — COMMENTED** [review]
Great job! Could you please only add the `;` at the end of the lines to follow the same pattern? Thanks!

**Review by davimacedo — APPROVED** [review]
LGTM!

**Comment by itzharDev** [comment]
great & tnx
when is the next release btw?

**Comment by davimacedo** [comment]
@mtrezza @dplewis I think we can release a new version right away. What do you guys think?

**Comment by mtrezza** [comment]
Agree, and by the way this is a clever PR.

**Comment by itzharDev** [comment]
hi, any new with the release date?

**Comment by mtrezza** [comment]
There is an issue resulting from this. When showing previously hidden columns, the fields are `(undefined)`, because their values are not fetched. Only when manually reloading the list, the values appear.

**Pull request merged**